File: SDKs/rev.ai/v1/typescript/src/sdk/models/operations/gettranscriptbyid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetTranscriptByIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetTranscriptByIdHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Accept" })
  accept?: shared.AcceptTranscriptEnum;
}


// GetTranscriptById404ApplicationProblemPlusJson
/** 
 * Problem details object returned on errors
**/
export class GetTranscriptById404ApplicationProblemPlusJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: number;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}


// GetTranscriptById406ApplicationProblemPlusJson
/** 
 * Problem details object returned on errors
**/
export class GetTranscriptById406ApplicationProblemPlusJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=allowed_values" })
  allowedValues?: string[];

  @SpeakeasyMetadata({ data: "json, name=current_value" })
  currentValue?: string;

  @SpeakeasyMetadata({ data: "json, name=detail" })
  detail?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: number;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}


// GetTranscriptById409ApplicationProblemPlusJson
/** 
 * Problem details object returned on errors
**/
export class GetTranscriptById409ApplicationProblemPlusJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=allowed_values" })
  allowedValues?: string[];

  @SpeakeasyMetadata({ data: "json, name=current_value" })
  currentValue?: string;

  @SpeakeasyMetadata({ data: "json, name=detail" })
  detail?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: number;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}


export class GetTranscriptByIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetTranscriptByIdPathParams;

  @SpeakeasyMetadata()
  headers: GetTranscriptByIdHeaders;
}


export class GetTranscriptByIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  getTranscriptById401ApplicationProblemPlusJsonAny?: any;

  @SpeakeasyMetadata()
  getTranscriptById404ApplicationProblemPlusJsonObject?: GetTranscriptById404ApplicationProblemPlusJson;

  @SpeakeasyMetadata()
  getTranscriptById406ApplicationProblemPlusJsonObject?: GetTranscriptById406ApplicationProblemPlusJson;

  @SpeakeasyMetadata()
  getTranscriptById409ApplicationProblemPlusJsonObject?: GetTranscriptById409ApplicationProblemPlusJson;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  transcript?: shared.Transcript;

  @SpeakeasyMetadata()
  transcriptText?: string;
}

File: SDKs/rev.ai/v1/python/src/sdk/models/operations/gettranscriptbyid.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import accepttranscript_enum as shared_accepttranscript_enum
from ..shared import transcript as shared_transcript


@dataclasses.dataclass
class GetTranscriptByIDPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetTranscriptByIDHeaders:
    accept: Optional[shared_accepttranscript_enum.AcceptTranscriptEnum] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Accept', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class GetTranscriptByID404ApplicationProblemPlusJSON:
    r"""GetTranscriptByID404ApplicationProblemPlusJSON
    Problem details object returned on errors
    """
    
    status: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclasses.dataclass
class GetTranscriptByID406ApplicationProblemPlusJSON:
    r"""GetTranscriptByID406ApplicationProblemPlusJSON
    Problem details object returned on errors
    """
    
    allowed_values: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('allowed_values') }})
    current_value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current_value') }})
    detail: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('detail') }})
    status: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclasses.dataclass
class GetTranscriptByID409ApplicationProblemPlusJSON:
    r"""GetTranscriptByID409ApplicationProblemPlusJSON
    Problem details object returned on errors
    """
    
    allowed_values: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('allowed_values') }})
    current_value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current_value') }})
    detail: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('detail') }})
    status: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclasses.dataclass
class GetTranscriptByIDRequest:
    headers: GetTranscriptByIDHeaders = dataclasses.field()
    path_params: GetTranscriptByIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetTranscriptByIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    get_transcript_by_id_401_application_problem_plus_json_any: Optional[Any] = dataclasses.field(default=None)
    get_transcript_by_id_404_application_problem_plus_json_object: Optional[GetTranscriptByID404ApplicationProblemPlusJSON] = dataclasses.field(default=None)
    get_transcript_by_id_406_application_problem_plus_json_object: Optional[GetTranscriptByID406ApplicationProblemPlusJSON] = dataclasses.field(default=None)
    get_transcript_by_id_409_application_problem_plus_json_object: Optional[GetTranscriptByID409ApplicationProblemPlusJSON] = dataclasses.field(default=None)
    transcript: Optional[shared_transcript.Transcript] = dataclasses.field(default=None)
    transcript_text: Optional[str] = dataclasses.field(default=None)
    
